source/agora/consensus/EnrollmentManager.d

```diff
@@ -439,6 +533,24 @@ public class EnrollmentManager
         return this.validator_set.getEnrolledUTXOs(keys);
     }
 
+    /***************************************************************************
+
+        Get all the enrolled validator's public keys.
+
+        Params:
+            keys = will contain the set of public keys
+            height = the height of proposed block
+
+        Returns:
+            Return true if there was no error in getting the public keys
+
+    ***************************************************************************/
+
+    public bool getActiveValidatorPublicKeys (out PublicKey[] keys, Height height) @safe nothrow
```

Comment:
This should be a `ref` and inside of `getActiveValidatorPublicKeys` the keys length should be set to 0 before calling `assumeSafeAppend`.

As I mentioned before, `out` will set the array to null. So `assumeSafeAppend` will not have the intended effect here.